### PR TITLE
Update Collection<T> type-hint to Collection<TKey, T> for ManyToMany relationship properties

### PR DESCRIPTION
**Pull request by jszutkowski**

### CODE REVIEW

File: rules/CodeQuality/Enum/ToManyMappings.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+declare(strict_types=1);
+
+namespace Rector\Doctrine\CodeQuality\Enum;
+
+class ToManyMappings
+{
+    /**
+     * @var class-string[]
+     */
+    public const TO_MANY_CLASSES = ['Doctrine\\ORM\\Mapping\\OneToMany', 'Doctrine\\ORM\\Mapping\\ManyToMany'];
```
Comment (jszutkowski) [inline rules/CodeQuality/Enum/ToManyMappings.php:12]: Extracted relation mapping classes to use them in multiple places not to duplicate code.

Initially I created two consts: ONE_TO_MANY_CLASS and MANY_TO_MANY_CLASS, but phpstan complained that they are not used anywhere despite they were used in TO_MANY_CLASSES

**Comment by jszutkowski** [comment]
All *.inc files were copied from already existing ones for OneToMany relationship and replaced with ManyToMany relationships

**Comment by jszutkowski** [comment]
There were problems with running checks because some dependencies require PHP 8.2 now. I updated workflows to use higher version of PHP so that the dependencies could be installed. Not sure if that should be done in this or a separate PR. Would be also good to update PHP version in composer.json

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you @jszutkowski :clap:

**Pull request merged**